lib/ansible/plugins/inventory/tower.py

```diff
@@ -0,0 +1,201 @@
+    authors:
+      - Matthew Jones (@matburt)
+      - Yunfan Zhang (@YunfanZhang42)
+    short_description: Ansible dynamic inventory plugin for Ansible Tower.
+    description:
+        - Imports inventories from Ansible Tower.
+        - Supports reading configuration from both YAML config file and environment variables.
+        - If reading from the YAML file, the file name must end with tower_inventory.(yml|yaml),
+          the path in the command would be /path/to/tower_inventory.(yml|yaml). If some arguments in the config file
+          are missing, this plugin will try to fill in missing arguments by reading from environment variables.
+        - If reading configurations from environment variables, the path in the command would be tower_inventory.
+    options:
+        tower_host:
+            description: The network address of your Ansible Tower host.
+            type: string
+            env:
+                - name: TOWER_HOST
+            required: True
+        tower_username:
+            description: The user that you plan to use to access inventories on tower.
+            type: string
+            env:
+                - name: TOWER_USERNAME
+            required: True
+        tower_password:
+            description: The password for tower_username.
+            type: string
+            env:
+                - name: TOWER_PASSWORD
+            required: True
+        tower_inventory:
+            description: The ID of the tower inventory you wish to import.
+            type: string
+            env:
+                - name: TOWER_INVENTORY
+            required: True
+        tower_verify_ssl:
+            description: Specify whether Ansible should verify the SSL certificate of Ansible tower.
+            type: bool
+            default: True
+            env:
+                - name: TOWER_VERIFY_SSL
+            required: False
+'''
+
+EXAMPLES = '''
+# Before you execute the following commands, you should make sure this file is in your plugin path,
+# and you enabled this plugin.
+
+# Example for using tower_inventory.yml file
+
+plugin: tower
+tower_host: your_tower_server_network_address
+tower_username: your_username
+tower_password: your_password
+tower_inventory: the_ID_of_targeted_inventory
+# Then you can run the following command.
+# If some of the arguments are missing, Ansible will attempt to read them from environment variables.
+# ansible-inventory -i /path/to/tower_inventory.yml --list
+
+# Example for reading from environment variables:
+
+# Set environment variables:
+# export TOWER_HOST=YOUR_TOWER_HOST_ADDRESS
+# export TOWER_USERNAME=YOUR_TOWER_USERNAME
+# export TOWER_PASSWORD=YOUR_TOWER_PASSWORD
+# export TOWER_INVENTORY=THE_ID_OF_TARGETED_INVENTORY
+# Read the inventory specified in TOWER_INVENTORY from Ansible Tower, and list them.
+# The inventory path must always be tower_inventory if you are reading all settings from environment variables.
+# ansible-inventory -i tower_inventory --list
```

Comment:
you can also just have a config file with `plugin: tower` as it's only entry

Reply:
Hey Brian thanks for your comments! The reason we did it this way is this plugin is adapted from an inventory script of Ansible Tower, so the we want to keep this "read environment variable only, do not require any config file" behavior to be backward compatible with the old way of reading inventory from Ansible Tower.

Reply:
Are other plugins supporting any kind of file-less use? It seems like that's more of where this was trying to go.

Reply:
No. They need to have a config file with `plugin: plugin_name` in it, and then read everything else from env var.

Reply:
my other issue is if user has an actual 'tower_inventory' file in cwd ... i would use something more like `@tower_inventory` or some other symbol to disambiguate from 'actual file'

Reply:
If they have an actual file, I think that file will end with `.yml` so this plugin will not mistakenly read it?